### PR TITLE
Add row and cell generator methods

### PR DESCRIPTION
**Pull request by assertchris**
The reason for this change is to allow row-level and cell-level changes in subclasses.

Also: https://github.com/silverstripe/silverstripe-framework/pull/4079/files#diff-c804f1dc153eaf30afb92594fa774456L362

### CODE REVIEW

File: forms/gridfield/GridField.php [hunk 1]
```diff
@@ -83,9 +87,9 @@ class GridField extends FormField {
 	/**
 	 * Creates a new GridField field
 	 *
-	 * @param string $name
-	 * @param string $title
-	 * @param SS_List $dataList
+	 * @param string          $name
```
Comment (tractorcow) [inline forms/gridfield/GridField.php]: Can we just use simple spacing for phpdoc? Most of the codebase doesn't follow aligned indentation.

**Comment by tractorcow** [comment]
I guess this doesn't really add much rather than refactor existing code. Do you think this is minor enough to fit into 3.1? cc @silverstripe/core-team ?

**Comment by tractorcow** [comment]
I would say "style sucks, so minor code cleanup" is safe to merge into a patch release.

**Comment by willmorgan** [comment]
I think that it's safe to go in, though I've gotta say all the reformatting makes it hard to review :wink:

Thanks @assertchris!

**Comment by dhensby** [comment]
@willmorgan I'm only 4 months late but you can add `?w=1` to the end of the url to see it without whitespace changes!

**Pull request merged**